parasoft-findings-teamcity-agent/src/test/java/com/parasoft/findings/teamcity/agent/ParasoftFindingsBuildProcessTest.java

```diff
@@ -0,0 +1,214 @@
+package com.parasoft.findings.teamcity.agent;
+
+import jetbrains.buildServer.agent.AgentRunningBuild;
+import jetbrains.buildServer.agent.BuildFinishedStatus;
+import jetbrains.buildServer.agent.BuildProgressLogger;
+import jetbrains.buildServer.agent.BuildRunnerContext;
+import jetbrains.buildServer.messages.BuildMessage1;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.*;
+
+import java.io.File;
+import java.text.DecimalFormat;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.powermock.api.mockito.PowerMockito.*;
+
+public class ParasoftFindingsBuildProcessTest {
+    @InjectMocks
+    ParasoftFindingsBuildProcess parasoftFindingsBuildProcess;
+    @Mock
+    AgentRunningBuild build;
+    @Mock
+    BuildRunnerContext context;
+    @Mock
+    BuildProgressLogger buildProgressLogger;
+
+    Map<String, String> params = new HashMap<>();
+    String reportDirPath = "src/test/resources/report";
+    File reportDir = new File(reportDirPath);
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void test_transformStaticTestReport_withLocalSettings() throws Throwable {
+        params.put("settings.location", "localSettings.properties");
+        File localSettings = new File(reportDirPath + "/localSettings.properties");
+
+        testTransformStaticTestReport();
+
+        Mockito.verify(buildProgressLogger).message("File path for local settings is " + localSettings.getAbsolutePath());
+    }
+
+    @Test
+    public void test_transformStaticTestReport_withEmptyLocalSettings() throws Throwable {
+        params.put("settings.location", "emptyLocalSettings.properties");
+        doNothing().when(buildProgressLogger).warning(Mockito.isA(String.class));
+
+        testTransformStaticTestReport();
+
+        Mockito.verify(buildProgressLogger, Mockito.atLeastOnce()).warning(Mockito.isA(String.class));
+        Mockito.verify(buildProgressLogger).warning("No properties loaded");
+    }
+
+
+    private void testTransformStaticTestReport() throws Throwable {
+        // Given
+        reportDirPath = reportDirPath + "/static";
+        params.put("reports.location", "static/cpptest_professional_report.xml");
+        File testReport = new File(reportDirPath + "/cpptest_professional_report.xml");
+        File pmdReport = new File(reportDirPath + "/pmd-cpptest_professional_report.xml");
+        File pmdCpdReport = new File(reportDirPath + "/pmdCpd-cpptest_professional_report.xml");
+        String pmdReportRelativePath = reportDir.toURI().relativize(pmdReport.toURI()).getPath();
+        prepareDataForTestTransformation();
+
+        doNothing().when(buildProgressLogger).message(Mockito.isA(String.class));
+        doNothing().when(buildProgressLogger).logMessage(Mockito.isA(BuildMessage1.class));
```

Comment:
This is to mock the function "_build.getBuildLogger().logMessage(DefaultMessagesInfo.createTextMessage(anyString))" in ParasoftFindingsBuildProcess.java. And the parameter value used by this function can be collected. But the value is a "BuildMessage1" entity. If to create a new entity to assert this, there is an error could be displayed, which is that assert value cannot be found. I have tried to mock "BuildMessage1" and return a value when creating a new entity. But it still  does not work. So I just assert the function has been called here.